lib/safira_web/authorize.ex

```diff
@@ -0,0 +1,29 @@
+defmodule Safira.Authorize do
+  import Plug.Conn
+  import Phoenix.Controller
+  
+  alias Safira.Repo
+  alias Safira.Accounts.User
+
+  def init(_params) do
+  end
+
+  def call(conn, _params) do
+    if is_authorized(conn) do
+      conn
+    else
+      conn
+      |> put_status(:unauthorized)
+      |> json(%{error: "Cannot access resource"})
+      |> halt()
+    end
+  end
+
+  defp is_authorized(conn) do
```

Comment:
i think we need to generalize this plug to define permissions for every time of user